### PR TITLE
[TASK4] added 4th blocks

### PR DESCRIPTION
**Pull request by NikitaKhadnevich**
Added tasks which work with async-code,
created html-wrapper for demo,
configure some css-style file for work

### CODE REVIEW

File: task4/index3.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+//---SubTask---В. 
+//Написать Promise который выводит console.log в случае если время выполнения не превысило 2 секунд, и console.error в случае если превысило. Время должно определяться случайным образом, использовать внутренний обработчик ошибок для Promise.//
+
+createError = () => error = new Error('Time out...');
+createSuccess = () => createSuccess = 'Well done!';
+generatorHelper = () => number = Math.floor(Math.random()*10)*1000;
```
Comment (k-krasilnikova) [inline task4/index3.js:6]: magic number!!!
Move it please to a separate constant with a readable name.

File: task4/index3.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+createError = () => error = new Error('Time out...');
+createSuccess = () => createSuccess = 'Well done!';
+generatorHelper = () => number = Math.floor(Math.random()*10)*1000;
+
+function generatorCheker(delay) {
+  console.log('Start Number Generator...')
+  console.log(delay)
+  let promise = new Promise((resolve, reject) => {
+     setTimeout(() => {
+        delay <= 2000 ? resolve() : reject()
```
Comment (k-krasilnikova) [inline task4/index3.js:13]: magic number!!!
Move it please to a separate constant with a readable name.

File: task4/index4.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+   constructor(response) {
+      super(`${response.status} for ${response.url}`);
+      this.name = 'HttpError';
+      this.response = response;
+   }
+}
+
+async function loadJson(url) {
+   let response = await fetch(url)
+   if (response.status == 200) {
```
Comment (k-krasilnikova) [inline task4/index4.js:16]: Let's create a separate constant with some http statuses to avoid using of magic numbers

File: task4/index4.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class HttpError extends Error {
+   constructor(response) {
+      super(`${response.status} for ${response.url}`);
+      this.name = 'HttpError';
+      this.response = response;
+   }
+}
+
+async function loadJson(url) {
+   let response = await fetch(url)
```
Comment (k-krasilnikova) [inline task4/index4.js:15]: Please don't forget to insert semicolumns at the end of the lines. Why `let` here?

File: task4/index4.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+   if (response.status == 200) {
+      const responses = await response.json()
+      console.log(responses)
+      return responses
+   } else {
+      throw new HttpError(response)
```
Comment (k-krasilnikova) [inline task4/index4.js:21]: Please don't forget to insert semicolumns at the end of the lines

File: task4/index4.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      super(`${response.status} for ${response.url}`);
+      this.name = 'HttpError';
+      this.response = response;
+   }
+}
+
+async function loadJson(url) {
+   let response = await fetch(url)
+   if (response.status == 200) {
+      const responses = await response.json()
```
Comment (k-krasilnikova) [inline task4/index4.js:17]: It's pretty hard to navigate between `responses` and `response`. Rename one of the variable/ or both of them to something more readable.

File: task4/index4.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+            alert("We can’t find such user."); // Add error if 404
+         } else { //Add error if otherwise
+            throw err
+         }
+      }
+   }
+alert(`Full name: ${user.name}`);
+return user // 'build user'
+}
+
```
Comment (k-krasilnikova) [inline task4/index4.js:43]: Semicolumns

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]
Some updates required!

File: task4/index3.js (hunk 1 above)
Comment (NikitaKhadnevich) [reply task4/index3.js:6]: Add another const and rebuild generatorHelper-fnc

File: task4/index3.js (hunk 2 above)
Comment (NikitaKhadnevich) [reply task4/index3.js:13]: Added new const

File: task4/index4.js (hunk 4 above)
Comment (NikitaKhadnevich) [reply task4/index4.js:15]: it was replace let to const

File: task4/index4.js (hunk 3 above)
Comment (NikitaKhadnevich) [reply task4/index4.js:16]: Added new const with value 200, replaced old value

File: task4/index4.js (hunk 6 above)
Comment (NikitaKhadnevich) [reply task4/index4.js:17]: Renamed responses to gitUsers

File: task4/index4.js (hunk 5 above)
Comment (NikitaKhadnevich) [reply task4/index4.js:21]: Added

File: task4/index4.js (hunk 7 above)
Comment (NikitaKhadnevich) [reply task4/index4.js:43]: Added again

**Review by k-krasilnikova — APPROVED** [review]

**Pull request merged**